File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/OpenApiIncludeTest.java
package io.micronaut.configuration.openapi.docs;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

class OpenApiIncludeTest {

    @Test
    void compiledControllersAreIncluded() {
        var paths = OpenApiSpec.load().getPaths();

        var login = paths.get("/login").getPost();
        assertEquals(List.of("Security"), login.getTags());
        assertNotNull(paths.get("/logout"));
        var env = paths.get("/env").getGet();
        assertEquals(List.of("Management"), env.getTags());
        assertEquals(List.of("ADMIN"), env.getSecurity().get(0).get("BEARER"));
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/openapi/docs/OpenApiIncludeTest.kt
package io.micronaut.configuration.openapi.docs

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test

class OpenApiIncludeTest {

    @Test
    fun compiledControllersAreIncluded() {
        val paths = OpenApiSpec.load().paths

        val login = paths["/login"]!!.post
        assertEquals(listOf("Security"), login.tags)
        assertNotNull(paths["/logout"])
        val env = paths["/env"]!!.get
        assertEquals(listOf("Management"), env.tags)
        assertEquals(listOf("ADMIN"), env.security[0]["BEARER"])
    }
}
